mwlwifi: Update to upstream version

thagabe opened the pull request:
Since last commit:
- Bug regarding older mwlwifi devices was identify and fix.
- Out-of-tree patch was upstreamed

Build and compiled: WRT3200ACM and WRT1900ACv1

Sign-off-by: Gabe Rodriguez \<lifehacksback@gmail.com\>

hauke commented:
Please squash these two patches together into one commit and add a Signed-of-by line to the commit message.

thagabe committed:
mwlwifi: Updated to upstream stable release

mwlwifi was updated to a new stable. Included in this stable release are the
followin benefits:
- Fixed compiling for kernel 4.14
- Fixed crash on 88W8864 binary

Compiled and tested on: WRT3200ACM and WRT1900AC

Signed-off-by: Gabe Rodriguez <lifehacksback@gmail.com>

thagabe commented:
@hauke 
Thanks for your patience! I have finally been able to squash the commits and added the signed-off.

hauke commented:
Thank you for the patch it was applied to master.
